lib/pwned_password_downloader.rb

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env ruby
+require 'fileutils'
+require 'net/http/persistent'
+require 'retries'
+require 'ruby-progressbar'
+
+class PwnedPasswordDownloader
+  attr_reader :destination,
+              :num_threads,
+              :keep_threshold
+
+  RANGE_API_ROOT = 'https://api.pwnedpasswords.com/range/'
+  SHA1_LENGTH = 40
+  HASH_PREFIX_LENGTH = 5
+  OCCURRENCE_OFFSET = SHA1_LENGTH - HASH_PREFIX_LENGTH + ':'.length
+
+  def initialize(
+    destination: 'tmp/pwned',
+    num_threads: 64,
+    keep_threshold: 30
+  )
+    @destination = destination
+    @num_threads = num_threads
+    @keep_threshold = keep_threshold
+  end
+
+  def run!(start: '00000', finish: 'FFFFF')
+    (start.to_i(16)..finish.to_i(16)).each do |prefix_num|
+      queue << prefix_num.to_s(16).upcase.rjust(5, '0')
```

Comment:
now that we have a constant
```suggestion
      queue << prefix_num.to_s(16).upcase.rjust(HASH_PREFIX_LENGTH, '0')
```

Reply:
> now that we have a constant

👍 Updated in 65a5e7b0e.